catroid/src/org/catrobat/catroid/content/Scene.java

```diff
@@ -0,0 +1,444 @@
+
+		Sprite background;
+		try {
+			background = new Sprite(context.getString(R.string.background));
+		} catch (Resources.NotFoundException e) {
+			//Because in test project we can't find the string
+			background = new Sprite("Hintergrund");
+		}
+		background.look.setZIndex(0);
+		addSprite(background);
+	}
+
+	public String getName() {
+		return sceneName;
+	}
+
+	public synchronized void addSprite(Sprite sprite) {
+		if (spriteList.contains(sprite)) {
+			return;
+		}
+		spriteList.add(sprite);
+	}
+
+	public synchronized boolean removeSprite(Sprite sprite) {
+		return spriteList.remove(sprite);
+	}
+
+	@Override
+	public Scene clone() {
+		Scene result = null;
+		try {
+			result = StorageHandler.getInstance().cloneScene(this);
+			result.correctUserVariableAndListReferences();
+			return result;
+		} catch (Exception e) {
+			Log.e("Scene.clone", e.getMessage());
+			return result;
+		}
+	}
+
+	public Scene cloneForBackPack() {
+		Scene clonedScene;
+		try {
+			ProjectManager.getInstance().checkNestingBrickReferences(false, true, true);
+			clonedScene = StorageHandler.getInstance().cloneScene(this);
+			if (!clonedScene.mergeProjectVariables()) {
+				return null;
+			}
+			clonedScene.correctUserVariableAndListReferences();
+		} catch (Exception e) {
+			Log.e("Scene.cloneForBackpack", e.getMessage());
+			return null;
+		}
+
+		return clonedScene;
+	}
+
+	public boolean mergeProjectVariables() {
+		ArrayList<String> variables = new ArrayList<>();
+		ArrayList<String> lists = new ArrayList<>();
+
+		for (Sprite sprite : spriteList) {
+			for (Brick brick : sprite.getListWithAllBricks()) {
+				if (brick instanceof UserVariableBrick && !variables.contains(((UserVariableBrick) brick)
+						.getUserVariable().getName())) {
+					variables.add(((UserVariableBrick) brick).getUserVariable().getName());
+				}
+				if (brick instanceof UserListBrick
+						&& !lists.contains(((UserListBrick) brick).getUserList().getName())) {
+					lists.add(((UserListBrick) brick).getUserList().getName());
+				}
+				if (brick instanceof FormulaBrick) {
+					for (Formula formula : ((FormulaBrick) brick).getFormulas()) {
+						formula.getVariableAndListNames(variables, lists);
+					}
+				}
+			}
+		}
+
+		for (String variable : variables) {
+			if (dataContainer.existVariableInAnySprite(variable, spriteList)) {
+				return false;
+			}
+			if (!dataContainer.existProjectVariableWithName(variable)) {
+				dataContainer.addProjectUserVariable(variable);
+			}
+		}
+
+		for (String list : lists) {
+			if (dataContainer.existListInAnySprite(list, spriteList)) {
+				return false;
+			}
+			if (!dataContainer.existProjectListWithName(list)) {
+				dataContainer.addProjectUserList(list);
+			}
+		}
+
+		return true;
+	}
+
+	@Override
+	public String toString() {
+		return sceneName;
+	}
+
+	public int getOriginalWidth() {
+		return originalWidth;
+	}
+
+	public int getOriginalHeight() {
+		return originalHeight;
+	}
+
+	public List<Sprite> getSpriteList() {
```

Comment:
we should not forget to adapt this for my clone feature ( PR #1823 ) after this or my feature has been merged ;)